MIKKI-61

vdvhk12 opened the pull request:
<!--
  템플릿은 아직 PR 작성이 익숙하지 않으신 분들을 위해서 제공하는 가이드입니다!
  리뷰어 또는 이 PR을 보게 될 다른 사람들이 이 PR을 보는데 참고할 수 있는 내용이 있다면 포함해서 작성해주시면 됩니다.
-->

## 📌 과제 설명 <!-- 어떤 걸 만들었는지 대략적으로 설명해주세요 -->
- 이미지 업로드 S3 Config설정

## 👩‍💻 요구 사항과 구현 내용 <!-- 기능을 Commit 별로 잘개 쪼개고, Commit 별로 설명해주세요 -->
- S3 설정
- PresignedUrl 제공 API 구현

<!--
## ✅ 피드백 반영사항  <!-- 지난 코드리뷰에서 고친 사항을 적어주세요. 재PR 시에만 사용해 주세요! (재PR 아닌 경우 삭제) -->
<!--
## ✅ PR 포인트 & 궁금한 점 <!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->

vdvhk12 committed:
chore: .gitignore에 .idea, db 파일 추가

vdvhk12 committed:
chore: s3 의존성 추가 & S3 Config 설정 (MIKKI-61)

vdvhk12 committed:
feat: S3StorageService 구현 (MIKKI-61)

vdvhk12 committed:
feat: StorageController 구현 (MIKKI-61)

vdvhk12 committed:
refactor: 파일명 중복을 위한 파일명 생성 로직 추가 & Response dto 생성 (MIKKI-61)

vdvhk12 committed:
feat: 커스텀 예외 추가 (MIKKI-61)

Janghyeonsuk commented on backend/src/main/java/com/backend/global/storage/exception/StorageErrorCode.java:
```diff
@@ -0,0 +1,19 @@
+@RequiredArgsConstructor
+public enum StorageErrorCode {
+
+	//공통 서버 에러 코드 500
```
주석이 글로벌 에러로 되어있는거 같아 수정해주시면 좋을거같습니다.

vdvhk12 replied:
넵 확인했습니다~!

kangjjs commented on backend/src/main/java/com/backend/global/storage/service/S3StorageService.java:
저 file_size도 그냥 yml로 빼도 좋을거 같습니다

vdvhk12 committed:
chore: 에러 코드 주석 수정 (MIKKI-61)

vdvhk12 committed:
refactor: 파일 최대 크기 yml로 관리, 주석 추가 (MIKKI-61)

vdvhk12 committed:
docs: presigned-urls API 문서 추가 (MIKKI-61)

vdvhk12 committed:
chore: TODO 주석 추가 (MIKKI-61)

vdvhk12 committed:
refacotr: FileUploadRequest에 Validation 적용 (MIKKI-61)

- Request에 List로 파일을 받고 도메인도 쿼리스트링 -> Request 객체안에 포함하도록 수정
- 파일 크기는 Validation으로 검증하므로 커스텀 예외 처리에서 제외

vdvhk12 committed:
chore: 주석 위치 인터페이스로 수정 (MIKKI-61)

vdvhk12 committed:
refactor: 파일크기 에러코드 제거 (MIKKI-61)

- 컨트롤러에서 Validation으로 파일 크기 검증

vdvhk12 committed:
chore: 불필요한 주석 제거 (MIKKI-61)

vdvhk12 committed:
chore: 메서드에 implSpec 주석 제거 (MIKKI-61)
